code/datums/components/mood.dm

```diff
@@ -188,9 +204,94 @@
 			setSanity(sanity+0.4, SANITY_NEUTRAL, SANITY_MAXIMUM)
 		if(9)
 			setSanity(sanity+0.6, SANITY_NEUTRAL, SANITY_MAXIMUM)
+
+	var/psych_adjustment = round((sanity-SANITY_MAXIMUM/2)/SANITY_MAXIMUM + (mood + 2.5)/10)
+	/*What the fuck is this expression?
+		Basically. first get the sanity which is a value between 0 and 150. we subtract 75 from the sanity to get a value that is between -75 to 75
+		Secondly we take mood whcih is a value between 15 and -20 so we add 2.5 to it getting a value between 17.5 and -17.5.
+		We add both of them up and divide by ten to get a value between 9.25 and -9.25. We round it so we get an integer value
+	*/
+
+	adjustPsychInstability(psych_adjustment)
+	CheckPsychInstability()
 	HandleNutrition()
 
-///Sets sanity to the specified amount and applies effects.
+
+
+#define PSYCH_BREAK_EXTREME 100
+#define PSYCH_BREAK 50 to 99
+
+#define PSYCH_CURE_EXTREME -100
+#define PSYCH_CURE -50 to -99
+
+/datum/component/mood/proc/adjustPsychInstability(amount)
+	psych_instab = CLAMP(psych_instab - amount,  PSYCH_CURE_EXTREME, PSYCH_BREAK_EXTREME)
```

Comment:
Why the fuck did you make this proc SUBTRACT the parameter from the subject's psych instability value instead of ADD the parameter to the subject's psych instability value? This is VERY counter-intuitive.